### PR TITLE
feat(v2): allow activeBaseTest in NavLink

### PR DESCRIPTION
**Pull request by nebrelbug**
## Motivation

See #2683 and #2574 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Serving the Docusaurus v2 site works as expected. Adding `activeBaseTest: 'docs\\/(next|2\\.0\\.0\\-alpha\\.50)'` to the "Docs" link in `docusaurus.config.js` results in the "Docs" link only being highlighted on the "next" and "alpha.50" versions.

### CODE REVIEW

File: website/docs/theme-classic.md [hunk 1]
```diff
@@ -137,7 +137,9 @@ module.exports = {
           position: 'left', // or 'right'
           // To apply the active class styling on all
           // routes starting with this path.
+          // This usually isn't necessary
           activeBasePath: 'docs',
+          // activeBaseTest: 'docs/(next|v8)'
```
Comment (lex111) [inline website/docs/theme-classic.md:142]: Could we rename the option "activeBaseTest" with "activeBaseRegex"? It looks more understandable.

File: packages/docusaurus-theme-classic/src/theme/Navbar/index.js [hunk 2]
```diff
@@ -43,10 +44,16 @@ function NavLink({
...
             activeClassName,
             to: toUrl,
-            ...(activeBasePath
+            ...(activeBasePath || activeBaseTest
               ? {
-                  isActive: (_match, location) =>
-                    location.pathname.startsWith(activeBaseUrl),
+                  isActive: (_match, location) => {
+                    if (activeBaseTest) {
+                      const activeBaseReg = new RegExp(activeBaseTest);
```
Comment (lex111) [inline packages/docusaurus-theme-classic/src/theme/Navbar/index.js:51]: If we rename the option to `activeBaseRegex`, we can completely refuse from separate const.

```js
if (activeBaseRegex) {
  return new RegExp(activeBaseRegex).test(location.pathname);
}
```

It is OK?

Or we might use ternary expression at all, like that:

```js
isActive: (_match, location) => activeBaseTest ?
  new RegExp(activeBaseRegex).test(location.pathname) : 
  location.pathname.startsWith(activeBaseUrl)
```

File: website/docs/theme-classic.md [hunk 3]
```diff
@@ -148,6 +150,8 @@ module.exports = {
 };
 ```
 
+React Router should automatically apply active link styling to links, but you can use `activeBasePath` in edge cases. For cases in which a link should be active on several different paths (such as when you have multiple doc folders under the same sidebar), you can use `activeBaseTest`. `activeBaseTest` overrides `activeBasePath` -- Docusaurus parses it into a Regular Expression that is tested against the current URL.
```
Comment (lex111) [inline website/docs/theme-classic.md:153]:
```suggestion
React Router should automatically apply active link styling to links, but you can use `activeBasePath` in edge cases. For cases in which a link should be active on several different paths (such as when you have multiple doc folders under the same sidebar), you can use `activeBaseTest`. `activeBaseTest` overrides `activeBasePath` -- Docusaurus parses it into a regular expression that is tested against the current URL.
```

File: packages/docusaurus-theme-classic/src/theme/Navbar/index.js (hunk 2 above)
Comment (nebrelbug) [reply packages/docusaurus-theme-classic/src/theme/Navbar/index.js:51]: That makes a lot more sense! I'll change it to the ternary.

File: website/docs/theme-classic.md [hunk 4]
```diff
@@ -148,6 +150,8 @@ module.exports = {
 };
 ```
 
+React Router should automatically apply active link styling to links, but you can use `activeBasePath` in edge cases. For cases in which a link should be active on several different paths (such as when you have multiple doc folders under the same sidebar), you can use `activeBaseRegex`. `activeBaseRegex` overrides `activeBasePath` -- Docusaurus parses it into a regular expression that is tested against the current URL.
```
Comment (lex111) [inline website/docs/theme-classic.md:153]: > `activeBaseRegex` overrides `activeBasePath` 

I think it is necessary to reformulate here, something like this:

```suggestion
React Router should automatically apply active link styling to links, but you can use `activeBasePath` in edge cases. For cases in which a link should be active on several different paths (such as when you have multiple doc folders under the same sidebar), you can use `activeBaseRegex`. `activeBaseRegex` is more flexible alternative to `activeBasePath` and takes precedence over it -- Docusaurus parses it into a regular expression that is tested against the current URL.

File: website/docs/theme-classic.md [hunk 5]
```diff
@@ -137,7 +137,9 @@ module.exports = {
           position: 'left', // or 'right'
           // To apply the active class styling on all
           // routes starting with this path.
+          // This usually isn't necessary
           activeBasePath: 'docs',
+          // activeBaseRegex: 'docs/(next|v8)'
```
Comment (lex111) [inline website/docs/theme-classic.md:142]: We must explicitly mention that these two options are related, like that:

```suggestion
          activeBaseRegex: 'docs/(next|v8)', // Alternatively to above one
```

**Comment by nebrelbug** [comment]
Any chance this could be merged? I'm looking forward to using this in my documentation website :)

**Comment by lex111** [comment]
friendly ping @yangshun

**Comment by yangshun** [comment]
Thanks @nebrelbug! Much appreciated!

**Pull request merged**